src/utils/history.js

```diff
@@ -305,9 +309,28 @@ export const getHistoryEventsFromTransactions = (
     historyEvent = { ...historyEvent, ensName: extra.ensName };
   }
 
+  historyEvent = { ...historyEvent, type: eventType };
+
+  if (addressesEqual(fromAddress, activeAccountAddress)) {
```

Comment:
Aren't we going to add this deployment event before every tx originating from active account?

Reply:
Updated this with check if successful or pending event is added. It will keep adding failed outgoing until adds successful or pending.